### PR TITLE
Remove use of RTTI in CUDA provider

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: These usages are downcasts/upcasts that can be done without RTTI.

I'm running the zip packaging pipeline here: https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=128375&view=results to ensure it's working.

### CODE REVIEW

File: onnxruntime/core/providers/armnn/nn/pool.h [hunk 1]
```diff
@@ -22,7 +22,7 @@ class Pool final : public onnxruntime::Pool<T, PoolType> {
  public:
   explicit Pool(const OpKernelInfo& info) : onnxruntime::Pool<T, PoolType>(info) {
     provider_ = (const_cast<ArmNNExecutionProvider*>(
-        dynamic_cast<const ArmNNExecutionProvider*>(info.GetExecutionProvider())));
```
Comment (yuslepukhin) [inline onnxruntime/core/providers/armnn/nn/pool.h:25]: >dynamic_c [](start = 8, length = 9)

I have written my proposal in the email.

File: onnxruntime/core/providers/acl/nn/batch_norm.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ class BatchNorm final : public OpKernel {
     ORT_ENFORCE(st.IsOK(), st.ErrorMessage());
 
     provider_ = (const_cast<ACLExecutionProvider*>(
```
Comment (yuslepukhin) [inline onnxruntime/core/providers/acl/nn/batch_norm.h:40]: Oh, we have been throught const_cast already.  Last time it required several days of debugging. Can you file a bug?

**Comment by faxu** [comment]
@tiagoshibata Can you please make the changes based on @yuslepukhin's suggestions before merging?

File: onnxruntime/core/providers/acl/nn/batch_norm.h (hunk 2 above)
Comment (tiagoshibata) [reply onnxruntime/core/providers/acl/nn/batch_norm.h:40]: Issue here or aiinfra?

Comment (tiagoshibata) [reply onnxruntime/core/providers/acl/nn/batch_norm.h:40]: Done (internal bug 809210)

**Pull request merged**